src/main/java/org/folio/helper/PurchaseOrderHelper.java

```diff
@@ -273,25 +273,17 @@ public Future<Void> updateOrder(CompositePurchaseOrder compPO, boolean deleteHol
   public Future<Void> handleFinalOrderStatus(CompositePurchaseOrder compPO, String initialOrdersStatus,
                                                         RequestContext requestContext) {
     PurchaseOrder purchaseOrder = convertToPurchaseOrder(compPO);
-    Promise<List<PoLine>> promise = Promise.promise();
-
-    if (isEmpty(compPO.getCompositePoLines())) {
-      purchaseOrderLineService.getPoLinesByOrderId(compPO.getId(), requestContext)
-        .onSuccess(promise::complete)
-        .onFailure(promise::fail);
-    } else {
-      Future.succeededFuture()
-        .map(v -> {
-          List<PoLine> poLines = PoLineCommonUtil.convertToPoLines(compPO.getCompositePoLines());
-          if (initialOrdersStatus.equals(compPO.getWorkflowStatus().value()))
```

Comment:
this if statement was a root cause, it prevented to this transitions from Open to Closed status